src/js/utils/JSONUtil.js

```diff
@@ -224,6 +239,12 @@ module.exports = {
           } else {
             arrayIndex = -1;
           }
+
+          // Update the ending position of the last block token on stack
+          let lastBlockToken = blockTokens.pop();
```

Comment:
Because if you open two or more block tokens, you will loose track of them since you will always look at the last one :wink: (Unless I got the question wrong)